Orange/data/table/base.py

```diff
@@ -0,0 +1,1409 @@
+        for var in chain(self.domain.variables, self.domain.metas):
+            if isinstance(var, TimeVariable):
+                self[var.name] = var.column_to_datetime(self[var.name])
+
+    def save(self, filename):
+        """Save a Table to a file.
+
+        Parameters
+        ----------
+        filename : str
+            The destination file path, absolute or relative.
+        """
+        ext = os.path.splitext(filename)[1]
+        from Orange.data.io import FileFormat
+        writer = FileFormat.writers.get(ext)
+        if not writer:
+            desc = FileFormat.names.get(ext)
+            if desc:
+                raise IOError("Writing of {}s is not supported".format(desc.lower()))
+            else:
+                raise IOError("Unknown file name extension.")
+        writer.write_file(filename, self)
+
+    def __getitem__(self, item):
+        # if selecting a column subset, we need to transfer weights so they don't just disappear
+        # only do this for multiple column selection, which returns a DataFrame by contract
+        if isinstance(item, (Sequence, pd.Index, np.ndarray)) and not isinstance(item, str) \
+                and all(isinstance(i, str) for i in item) \
+                and any(ic not in item for ic in self._INTERNAL_COLUMN_NAMES):
+            item = list(item) + [ic for ic in self._INTERNAL_COLUMN_NAMES if ic not in item]
+        return super().__getitem__(item)
+
+    def __setitem__(self, key, value):
+        # if the table has an empty index and we're inserting a the first row,
+        # the index would be created by pandas automatically.
+        # we want to maintain unique indices, so we override the index manually.
+        # we also need to set default weights, lest they be NA
+        new_index_and_weights = len(self.index) == 0
+
+        # SOMEWHAT PANDAS CONTRACT BREAKAGE:
+        # the pandas default behaviour when adding a new column as a Series is that
+        # the indexes are inner-joined: only the elements at the indices that exist
+        # in the current table are actually set, other elements are NA
+        # for easier handling of Orange behaviour, we effectively ignore the index
+        # on the series if the series is of the same length (like when adding a new column)
+        # to just merge the column into the table
+        if isinstance(value, pd.Series) and len(value) == len(self) and not new_index_and_weights:
+            value.index = self.index
+        super().__setitem__(key, value)
+
+        if new_index_and_weights:
+            new_id = self._new_id(len(self), force_list=True)
+            self.index = new_id
+            # super call because we'd otherwise recurse back into this
+            super().__setitem__(self._WEIGHTS_COLUMN, 1)
+
+    def __iter__(self):
+        """Iterate over the rows of this TableBase as SeriesBase, breaking the pandas contract.
```

Comment:
So obviously I am totally against this. Sure, we have to write
```py
for index, row_series in table.iterrows():
   ...
```
or
```py
for i in range(len(table)):
    row_series = table.iloc[i]
```
to iterate over rows as series. But this is a relatively slow operation in pandas; it's **great** that it's verbose! It won't give false pretenses and will hint the user toward looking into vectorizing oprations, using `.itertuples()` (much faster), or extracting `.X/values` first.

That's before even saying user contracts breakage should not be taken lightly.

In other words, there is _not a single good reason_ for this override.